tools/ci_build/github/azure-pipelines/templates/stages/mac-ios-packaging-build-stage.yml

```diff
@@ -117,16 +117,70 @@ stages:
         useXcpretty: false  # xcpretty can hide useful error output so we will disable it
       displayName: 'Build App Center iPhone arm64 tests'
 
+    - script: |
+        zip -r --symlinks $(Build.ArtifactStagingDirectory)/package_tests.zip ios_package_testUITests-Runner.app
+      workingDirectory: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/DerivedData/Build/Products/Debug-iphoneos'
+      displayName: "Create .zip file of the tests"
+
+    - script: |
+        python $(Build.SourcesDirectory)/tools/python/generate_plist.py --dest_file "exportOptions.plist"
+      workingDirectory: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/'
+      displayName: "Generate .plist file for the .ipa file"
+      env:
+        PROVISIONING_PROFILE_UUID: '$(APPLE_PROV_PROFILE_UUID)'
+        APPLE_TEAM_ID: '$(APPLE_TEAM_ID)'
+
+    - task: Xcode@5
+      inputs:
+        actions: 'archive'
+        xcWorkspacePath: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/apple_package_test.xcworkspace'
+        packageApp: true
+        archivePath: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/'
+        exportOptions: 'plist'
+        exportOptionsPlist: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/exportOptions.plist'
+        configuration: 'Debug'
+        sdk: 'iphoneos'
+        scheme: 'ios_package_test'
+        xcodeVersion: 'specifyPath'
+        xcodeDeveloperDir: '/Applications/Xcode_${{ variables.xcodeVersion }}.app/Contents/Developer'
+        args: '-derivedDataPath $(Build.BinariesDirectory)/app_center_test/apple_package_test/DerivedData'
+        workingDirectory: '$(Build.BinariesDirectory)/app_center_test/apple_package_test/'
+        useXcpretty: false
+      displayName: 'Create archive for the .ipa file'
+
+    # Use script step because exporting the .ipa file using the Xcode@5 task was too brittle (Xcode@5 is designed
+    # to handle both the .xcarchive step and the .ipa step in the same step -- ran into countless issues with signing
+    # and the .plist file)
```

Comment:
I'm completely fine with directly calling xcodebuild in script steps. the Xcode steps seem to add extra complexity.

do you think it would be simpler to directly run the corresponding xcodebuild commands from the other Xcode steps too?

Reply:
I'll give a try -- I think I originally found using the xcodebuild commands a little tricky to work with.

Reply:
Ah, I ran into similar errors to what I was running into when I was trying to reproduce the steps in the terminal locally... so I will leave the pipeline yaml as a mismash of Xcode task & xcodebuild script steps 😅

This was the error: "/Users/runner/work/1/b/app_center_test/apple_package_test/apple_package_test.xcodeproj: error: ios_package_test has conflicting provisioning settings. ios_package_test is automatically signed, but provisioning profile iOS Team Provisioning Profile has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor, or switch to manual signing in the Signing & Capabilities editor. (in target 'ios_package_test' from project 'apple_package_test')" 

even though I attempted to manually sign in the build-for-testing task.